traits/trait_types.py

```diff
@@ -3675,10 +3694,27 @@ def __init__(self, **metadata):
     def validate(self, object, name, value):
         """ Raises an error, since no values can be assigned to the trait.
         """
-        raise TraitError(
-            "The '%s' trait of %s instance is a read-only "
-            "UUID." % (name, class_of(object))
-        )
+        if not self.can_init:
```

Comment:
Building off of this, I'd add an explicit `can_init=False` to the arguments of the `__init__` method even if you then simply pass it straight through as metadata.  Having significant arguments in the signature is useful for software introspection tools.

Reply:
Sure, I moved it to a class attribute rather than putting in the metadata: https://github.com/enthought/traits/pull/459/commits/bef015d7ba065e2d51d898dfa079c05757c5a683